client-js/app/elements/labrad-registry.ts

```diff
@@ -61,31 +53,49 @@ export class LabradRegistry extends polymer.Base {
     return item.name.match(this.regex);
   }
 
-  selectKey() {
-    this.selDir = null;
-    this.selectType = 'key';
+  @listen('idx-changed')
+  changedSelected() {
+    console.log('ChangeSelected', this.idx);
   }
 
-  selectDir() {
-    this.selKey = null;
-    this.selectType = 'dir';
+  @computed()
+  selected(idx: number): boolean {
+    console.log("computing selected is: ", idx);
+    if (idx != null) {
+        return false;
+      } else {
+        console.log('doesnt exist');
+        return true;
+      }
+  }
+  @property({computed: '_computeSelectedType(idx)'})
+  selectedType: string;
+  
+  @property({computed: '_computeSelectedItem(idx)'})
+  selectedItem: string;
+  
+  _computeSelectedType(idx: number): string {
+    if (typeof this.dirs !== 'undefined') { //I don't know if there's a better way to do this
+      if (idx < this.dirs.length) {
```

Comment:
Ah thank you, I didn't realize you could do that in js. I knew it worked in python.